std/assembly/typedarray.ts

```diff
@@ -61,11 +61,12 @@ export class Int8Array extends ArrayBufferView {
     return LAST_INDEX_OF<Int8Array, i8>(this, searchElement, fromIndex);
   }
 
-  fill(value: i32, start: i32 = 0, end: i32 = i32.MAX_VALUE): Int8Array {
-    return FILL<Int8Array, i8>(this, value, start, end);
+  fill(value: i32, start: i32 = 0, end: i32 = i32.MAX_VALUE): this {
+    FILL<u8>(this.dataStart, this.length, u8(value), start, end);
+    return this;
   }
 
-  sort(comparator: (a: i8, b: i8) => i32 = COMPARATOR<i8>()): Int8Array {
+  sort(comparator: (a: i8, b: i8) => i32 = COMPARATOR<i8>()): this {
```

Comment:
Using a `: this` return here will infer differently in TS if the class is overloaded. As such, `: Int8Array` is somewhat more correct, regardless of the value returned.

Reply:
Ah, okey

Reply:
Btw should I refactor all such cases in Array and StaticArray as well?

Reply:
If the class can be overloaded, probably. In `StaticArray` it's not technically necessary I think because it is `@final` anyhow.

Reply:
Ok, anyway I refactored to explicit type everywhere